Set/Game/GameViewModel.swift

```diff
@@ -43,8 +100,20 @@ class GameViewModel: ObservableObject {
         let matchStatus = gameModel.toggleCard(by: chousenCard.id)
         if matchStatus == .successfulMatch || matchStatus == .unsuccessfulMatch {
             DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
-                self?.gameModel.finishTurn(for: matchStatus)
+                self?.gameModel.finishTurn(for: matchStatus, player: player)
+                self?.cleanUp()
             }
         }
     }
+    
+    private func cleanUp() {
+        whoseTurn = nil
+        timer?.invalidate()
+        timerTitle = ""
+        countDownTimer = 10
+    }
+    
+    func didSelect(player: Player) {
+        whoseTurn = player
```

Comment:
it seems that you want not only change whoseTurn, but here must be all logic what happens when somebody tap on a player

Reply:
yes, I also start timer in this method